src/etc/rc.newwanipv6

```diff
@@ -121,6 +121,26 @@ system_hosts_generate();
 
 $cacheip = @file_get_contents($cacheip_file);
 
+// If on DS-Lite, try to find a valid / usable AFTR address (either from DHCPv6 or manually configured)
+if (is_dslite_enabled($interface)) {
+    // Try to read dhcp option 64 (AFTR name) from the environment
+    $aftr_addr = trim(getenv("raw_dhcp_option_64"));
```

Comment:
very cool solution :)

Reply:
I just extended the raw option patch which is already around ;-)